src/commands/fetch/mod.rs

```diff
@@ -3,24 +3,25 @@ pub use super::*;
 pub mod fetch;
 pub mod setfetch;
 pub use fetch::*;
+use serde::{Deserialize, Serialize};
 pub use setfetch::*;
+use std::fmt;
 
 /// convert the field-value into the desired format.
 /// Returns `None` if the string is empty, as empty values must not be included in embeds.
-pub fn format_fetch_field_value(field_name: &str, value: String) -> Option<String> {
-    if value.is_empty() {
-        None
-    } else {
-        match field_name {
-            MEMORY_KEY => {
-                if value == "0" {
-                    None
-                } else {
-                    Some(format_bytes(&value))
-                }
+pub fn format_fetch_field_value(field_name: FetchField, value: String) -> Option<String> {
```

Comment:
the field_name can be taken by reference (as a `&FetchField`) her, allowing you to remove the `.clone()`s when calling this